client/shaders/nodes_shader/opengl_fragment.glsl

```diff
@@ -14,13 +15,43 @@ varying vec3 eyeVec;
 varying vec3 tsEyeVec;
 varying vec3 lightVec;
 varying vec3 tsLightVec;
+varying float nightRatio;
 
 bool normalTexturePresent = false;
 
 const float e = 2.718281828459;
 const float BS = 10.0;
 const float fogStart = FOG_START;
-const float fogShadingParameter = 1 / ( 1 - fogStart);
+float fogShadingParameter = 1 / ( 1 - fogStart);
+
+#ifdef ENABLE_FOG_NOISE
+
+// Simple, fast noise function.
+// See: https://gist.github.com/patriciogonzalezvivo/670c22f3966e662d2f83
+vec4 perm(vec4 x){return mod(((x * 34.0) + 1.0) * x, 289.0);}
```

Comment:
```suggestion
vec4 perm(vec4 x) { return mod(((x * 34.0) + 1.0) * x, 289.0); }
```

Reply:
All that copied code has horrible code style.

Reply:
Fixed